spring-cloud-data-module-deployers/spring-cloud-data-module-deployer-yarn/src/main/java/org/springframework/cloud/data/module/deployer/yarn/YarnModuleDeployer.java

```diff
@@ -0,0 +1,213 @@
+		appProperties.setProperty(PREFIX + "operation", "CLUSTERINFO");
+		appProperties.setProperty(PREFIX + "applicationId", yarnApplicationId);
+		appProperties.setProperty(PREFIX + "verbose", "false");
+		appProperties.setProperty(PREFIX + "clusterId", clusterId);
+		app.appProperties(appProperties);
+		String info = app.run();
+		logger.info("Output from YarnContainerClusterApplication run for CLUSTERINFO: " + info);
+
+		boolean deployed = false;
+		String[] lines = info.trim().split("\\r?\\n");
+		if (lines.length == 3 && lines[2].contains("RUNNING")) {
+			deployed = true;
+		}
+
+		YarnModuleInstanceStatus status = new YarnModuleInstanceStatus(id.toString(), deployed, null);
+		return ModuleStatus.of(id).with(status).build();
+	}
+
+	@Override
+	public Map<ModuleDeploymentId, ModuleStatus> status() {
+		HashMap<ModuleDeploymentId, ModuleStatus> statuses = new HashMap<ModuleDeploymentId, ModuleStatus>();
+		for (String clusterId : findRunningContainerClusters()) {
+			ModuleDeploymentId moduleDeploymentId = convertToModuleDeploymentId(clusterId);
+			statuses.put(moduleDeploymentId, status(moduleDeploymentId));
+		}
+		return statuses;
+	}
+
+	private static String findRunningXdYarnApp() {
+		YarnInfoApplication app = new YarnInfoApplication();
+		Properties appProperties = new Properties();
+		appProperties.setProperty("spring.yarn.internal.YarnInfoApplication.operation", "SUBMITTED");
+		appProperties.setProperty("spring.yarn.internal.YarnInfoApplication.verbose", "false");
+		appProperties.setProperty("spring.yarn.internal.YarnInfoApplication.type", "XD");
+		app.appProperties(appProperties);
+		String info = app.run();
+		logger.info("Full status response for SUBMITTED app " + info);
+
+		// TODO: either make this easier in YarnInfoApplication
+		//       or use rest api directly
+		String[] lines = info.split("\\r?\\n");
+		logger.info("Parsing application id from " + StringUtils.arrayToCommaDelimitedString(lines));
+		if (lines.length == 3) {
+			return lines[2].trim().split("\\s+")[0].trim();
+		} else {
+			return null;
+		}
+	}
+
+	private static Collection<String> findRunningContainerClusters() {
+		String yarnApplicationId = findRunningXdYarnApp();
+
+		YarnContainerClusterApplication app = new YarnContainerClusterApplication();
+		Properties appProperties = new Properties();
+		appProperties.setProperty(PREFIX + "operation", "CLUSTERSINFO");
+		appProperties.setProperty(PREFIX + "applicationId", yarnApplicationId);
+		app.appProperties(appProperties);
+		String info = app.run();
+		logger.info("Output from YarnContainerClusterApplication run for CLUSTERSINFO: " + info);
+
+		ArrayList<String> ids = new ArrayList<String>();
+		String[] lines = info.split("\\r?\\n");
+		if (lines.length > 2) {
+			for (int i = 2; i < lines.length; i++) {
+				ids.add(lines[i].trim());
+			}
+		}
+		return ids;
+	}
+
+	private static String convertFromModuleDeploymentId(ModuleDeploymentId moduleDeploymentId) {
```

Comment:
Not for this PR, but I'm wondering if we should define some kind of interface in the SPI to create runtime identifiers from `ModuleDeploymentId`...